components/AttributeAutocomplete/index.tsx

```diff
@@ -0,0 +1,96 @@
+import { SxProps } from '@mui/material'
+import { Autocomplete, Chip, TextField } from '@mui/material'
+import React from 'react'
+import { AttributeRequest, OptionsAttribute } from 'utils/types'
+
+interface AutocompleteAttributeOption {
+  id: string
+  label: string
+  value: string
+  color: string
+}
+
+interface Props {
+  attributes: OptionsAttribute[]
+  onChange?: (e: React.SyntheticEvent, attributes: AttributeRequest) => void
+  sx?: SxProps
+}
+
+function buildAutocompleteOptions(
+  attributes: OptionsAttribute[]
+): AutocompleteAttributeOption[] {
+  return attributes.reduce((acc, attribute) => {
+    const options = attribute.possibleValues.map((value) => ({
+      id: attribute._id!, // todo: remove when request and reponse objects are separated
+      label: attribute.name,
+      value,
+      color: attribute.color,
+    }))
+
+    return [...acc, ...options]
+  }, [] as AutocompleteAttributeOption[])
+}
+
+function buildAttributeRequest(
+  attributes: AutocompleteAttributeOption[]
+): AttributeRequest {
+  return attributes.reduce((acc, attribute) => {
+    const attrReq = {
+      [attribute.id]: attribute.value,
+    }
+
+    return { ...acc, ...attrReq }
+  }, {})
+}
+
+export default function AttributeAutocomplete({
+  attributes,
+  sx,
+  onChange,
+}: Props) {
+  const options = buildAutocompleteOptions(attributes)
+
+  return (
+    <Autocomplete
+      multiple
+      options={options}
+      groupBy={(option) => option.label}
+      getOptionLabel={(option) => option.value}
+      isOptionEqualToValue={(option, value) =>
+        option.value === value.value && option.label === value.label
+      }
+      renderTags={(tagValue, getTagProps) =>
+        tagValue.map((option, index) => (
+          <Chip
+            label={option.value}
+            style={{ backgroundColor: option.color }}
+            {...getTagProps({ index })}
+            key={option.id}
+          />
+        ))
+      }
+      renderInput={(params) => <TextField {...params} label="Attributes" />}
+      onChange={(e, attributes) => {
+        if (!attributes.length) {
+          if (!!onChange) onChange(e, {} as AttributeRequest)
```

Comment:
why have the event as an arg here?

Reply:
I kept this as an arg just in case there is a future use case where we need the event. I'm not sure it will ever come up but I think this keeps it more mechanistic; if the event is needed we don't need to change the implementation at all, and it's easy to not use the event.